### PR TITLE
[OPPRO-281] Support spark33 tpcds

### PR DESCRIPTION
**Pull request by jinchengchenghh**

### CODE REVIEW

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxTPCDSSuite.scala [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    ).map { table =>
+      val tablePath = new File(resourcePath, table).getAbsolutePath
+      val tableDF = spark.read.format(fileFormat).load(tablePath)
+      tableDF.createOrReplaceTempView(table)
+      (table, tableDF)
+    }.toMap
+  }
+
+  ignore("q7") {
+    val queryPath = "/mnt/DP_disk1/code/gluten/backends-velox/workload/tpcds/tpcds.queries.updated/"
```
Comment (rui-mo) [inline backends-velox/src/test/scala/io/glutenproject/execution/VeloxTPCDSSuite.scala:97]: May be better to use some relative path.

**Comment by rui-mo** [comment]
Verified locally on TPC-DS.

**Review by rui-mo — APPROVED** [review]

File: backends-velox/src/main/scala/org/apache/spark/shuffle/VeloxColumnarShuffleWriter.scala [hunk 2]
```diff
@@ -162,7 +161,8 @@ class VeloxColumnarShuffleWriter[K, V](
           .split(nativeSplitter, cb.numRows, cArray.memoryAddress())
         dep.splitTime.add(System.nanoTime() - startTime)
         dep.numInputRows.add(cb.numRows)
-        writeMetrics.incRecordsWritten(1)
+        // This metric is important, AQE use it to decide if EliminateLimit
+        writeMetrics.incRecordsWritten(cb.numRows())
         cArray.close()
```
Comment (FelixYBW) [inline backends-velox/src/main/scala/org/apache/spark/shuffle/VeloxColumnarShuffleWriter.scala:166]: @jinchengchenghh 
Can you add one metric named "input_batches" to count the number of record batch number? The metric is used in our eventlog analysis.

Comment (jinchengchenghh) [reply backends-velox/src/main/scala/org/apache/spark/shuffle/VeloxColumnarShuffleWriter.scala:166]: ok, I will start a new PR

**Pull request merged**